clang/test/Analysis/non_loc_compound.cpp

```diff
@@ -0,0 +1,17 @@
+// REQUIRES: crash-recovery, asserts
```

Comment:
Actually, that's a good point.

I added that bit because the only way I can *guarantee* it will crash (with main) is if we have asserts on.

I'll remove this, but if someone wants to check the test I added and they go "hey, this works on main!", then hopefully they'll try an assertion-enabled build 🙏